internal/topology/check/compatibility.go

```diff
@@ -74,9 +74,11 @@ func ObjectsAreInTheSameNamespace(current, desired client.Object) field.ErrorLis
 
 	// NOTE: this should never happen (webhooks prevent it), but checking for extra safety.
 	if current.GetNamespace() != desired.GetNamespace() {
-		allErrs = append(allErrs, field.Forbidden(
+		allErrs = append(allErrs, field.Invalid(
```

Comment:
What's the guideline for using invalid vs forbidden here?

Reply:
godoc:
```go
	// ErrorTypeInvalid is used to report malformed values (e.g. failed regex
	// match, too long, out of bounds).  See Invalid().
	ErrorTypeInvalid ErrorType = "FieldValueInvalid"
	// ErrorTypeForbidden is used to report valid (as per formatting rules)
	// values which would be accepted under some conditions, but which are not
	// permitted by the current conditions (such as security policy).  See
	// Forbidden().
	ErrorTypeForbidden ErrorType = "FieldValueForbidden"
```

As this case is not security related I personally would prefer field.Invalid (but also in other cases in this file to be consistent)

Reply:
I think the security is just an example here, there's other reasons e.g. business logic, for a field to be forbidden. The key idea here is 'as per formatting rules' but that would make most of our errors forbidden rather than invalid. I think we need to come up with some strong rubric for our webhooks.

Reply:
As for now we are using forbidden almost consistently for reporting not allowed transitions (e.g. from managed to managed) and this was the only exception.
So I changed it making it consistent with the same error we have e.g when validating templates

Reply:
Does this mean we should also change the other `field.Forbidden`'s in this file to `field.Invalid`?

I don't understand the difference between namespace must be the same vs kind or group. 

Or is it maybe that the namespace **must** always be equal vs. we might consider allowing changes to kind etc in the future and it's just our current policy to forbid that? (while it is permanent with the namespace)

Reply:
Fixed for namespace as well